helper/inet/inet.go

```diff
@@ -0,0 +1,81 @@
+package inet
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+const (
+	advertiseIp = iota
+	advertisePort
+)
+
+const (
+	address = iota
+	cidr
+)
+
+func AdvertiseIpFromSubnet(ip string) (string, error) {
+	// Return raw IP address if it is not in subnet-form
+	if !strings.Contains(ip, "/") {
+		return ip, nil
+	}
+
+	// Split subnet CIDR IP from the port
+	advertiseTokens := strings.Split(ip, ":")
+	if len(advertiseTokens) != 2 {
+		return "", errors.New(fmt.Sprintf("port was not given in advertise subnet: `%s`", ip))
```

Comment:
I missed the check directly above this block, so this actually still works. Sorry for the confusion!